File: branches/Medusa/Source/Fantasy.Jobs/JobPropertyGroup.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Collections;
using Fantasy.XSerialization;
using System.Xml;

namespace Fantasy.Jobs
{
    [XSerializable("properties", NamespaceUri=Consts.XNamespaceURI)] 
    [Serializable]
    public class JobPropertyGroup :  IConditionalObject, IEnumerable, ICloneable
    {
        [XArray(Serializer = typeof(JobPropertiesSerializer))]
        private List<JobProperty> _list = new List<JobProperty>();

        public JobProperty[] ToArray()
        {
            return this._list.ToArray();
        }

        public int Count
        {
            get { return this._list.Count; }
        }

        public JobProperty AddNewItem(string name)
        {
            JobProperty rs = new JobProperty() { Name = name };
            this._list.Add(rs);
            return rs;
        }

        public void Clear()
        {
            this._list.Clear();
        }

        public object Clone()
        {
            JobPropertyGroup rs = new JobPropertyGroup();
            rs.Condition = this.Condition;
            foreach (JobProperty prop in this._list)
            {
                this._list.Add((JobProperty)prop.Clone());
            }

            return rs;
        }

        public void RemoveItem(JobProperty itemToRemove)
        {
            this._list.Remove(itemToRemove);
        }

        public void RemoveAt(int index)
        {
            this._list.RemoveAt(index);
        }

        [XAttribute("condition")] 
        public string Condition { get; set; }

        public IEnumerator GetEnumerator()
        {
            return this._list.GetEnumerator();
        }
       
    }
}

File: Fantasy.Jobs/src/fantasy/jobs/JobPropertyGroup.java
package fantasy.jobs;

import java.io.*;

import fantasy.xserialization.*;

@XSerializable(name = "properties", namespaceUri=Consts.XNamespaceURI)
public class JobPropertyGroup implements IConditionalObject, Iterable<JobProperty>, Cloneable, Serializable
{

	/**
	 * 
	 */
	private static final long serialVersionUID = -848315926536200958L;
	@XArray(serializer = JobPropertiesSerializer.class, items = { })
	private java.util.ArrayList<JobProperty> _list = new java.util.ArrayList<JobProperty>();

	public final JobProperty[] toArray()
	{
		return this._list.toArray(new JobProperty[]{});
	}

	public final int getCount()
	{
		return this._list.size();
	}

	public final JobProperty AddNewItem(String name)
	{
		JobProperty tempVar = new JobProperty();
		tempVar.setName(name);
		JobProperty rs = tempVar;
		this._list.add(rs);
		return rs;
	}

	public final void Clear()
	{
		this._list.clear();
	}

	public final Object clone()
	{
		JobPropertyGroup rs = new JobPropertyGroup();
		rs.setCondition(this.getCondition());
		for (JobProperty prop : this._list)
		{
			this._list.add((JobProperty)prop.clone());
		}

		return rs;
	}

	public final void RemoveItem(JobProperty itemToRemove)
	{
		this._list.remove(itemToRemove);
	}

	public final void RemoveAt(int index)
	{
		this._list.remove(index);
	}

	@XAttribute(name = "condition")
	private String privateCondition;
	public final String getCondition()
	{
		return privateCondition;
	}
	public final void setCondition(String value)
	{
		privateCondition = value;
	}

	public final java.util.Iterator<JobProperty> iterator()
	{
		return this._list.iterator();
	}

}